File: 2024-03-28-intro-dev-tsjs/4.ts
import ollama from 'ollama';

const prompt = "why is the sky blue";

const output = await ollama.generate({ model: "llama2", prompt: "why is the sky blue?", stream: true })
const context: number[] = [];

for await (const part of output) {
  if (part.done === true) {
    console.log(`first generate complete`);
    context.push(...part.context);
  }
}

const output2 = await ollama.generate({ model: "llama2", prompt: "can it be another?", context: context });
console.log(`output with context\n\n${output2.response}\n\noutput complete\n`);


const output3 = await ollama.generate({ model: "llama2", prompt: "can it be another?" });
console.log(`output without context\n\n${output3.response}\n\noutput complete`);

File: 2024-04-01-intro-dev-python/4.py
import ollama 

prompt = "why is the sky blue";

output = ollama.generate(model="llama2", prompt=prompt, stream=True)

for chunk in output:
  if chunk["done"] == True:
    print("First Generate Complete")
    context = chunk["context"]  
    
output2 = ollama.generate(model="llama2", prompt="can it be another?", context=context);
# print(f"output with context\n\n{output2['response']}\n\noutput complete\n");


# output3 = ollama.generate(model="llama2", prompt="can it be another?");
# print(f"output without context\n\n{output3['response']}\n\noutput complete");
